src/NuGet.Core/NuGet.Commands/SourcesCommands/SourcesRunner.cs

```diff
@@ -0,0 +1,311 @@
+
+using System;
+using System.Globalization;
+using System.Linq;
+using NuGet.Common;
+
+namespace NuGet.Commands
+{
+    /// <summary>
+    /// Shared code to run the "sources" command from the command line projects
+    /// </summary>
+    public static class SourcesRunner
+    {
+        public static void Run(SourcesArgs args)
+        {
+            switch (args.Action)
+            {
+                case SourcesAction.List:
+                    switch (args.Format)
+                    {
+                        case CommandLine.SourcesListFormat.Short:
+                            PrintRegisteredSourcesShort(args);
+                            break;
+                        case CommandLine.SourcesListFormat.Detailed:
+                        default:
+                            PrintRegisteredSourcesDetailed(args);
+                            break;
+                    }
+                    break;
+                case SourcesAction.Add:
+                    AddNewSource(args);
+                    break;
+                case SourcesAction.Remove:
+                    RemoveSource(args);
+                    break;
+                case SourcesAction.Enable:
+                    EnableOrDisableSource(args, enabled: true);
+                    break;
+                case SourcesAction.Disable:
+                    EnableOrDisableSource(args, enabled: false);
+                    break;
+                case SourcesAction.Update:
+                    UpdatePackageSource(args);
+                    break;
+            }
+        }
+
+        private static void EnableOrDisableSource(SourcesArgs args, bool enabled)
+        {
+            if (string.IsNullOrEmpty(args.Name))
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandNameRequired));
+            }
+
+            var packageSource = args.SourceProvider.GetPackageSourceByName(args.Name);
+            if (packageSource == null)
+            {
+                args.LogError(string.Format(CultureInfo.CurrentCulture,
+                    Strings.SourcesCommandNoMatchingSourcesFound, args.Name));
+            }
+
+            if (enabled && !packageSource.IsEnabled)
+            {
+                args.SourceProvider.EnablePackageSource(args.Name);
+            }
+            else if (!enabled && packageSource.IsEnabled)
+            {
+                args.SourceProvider.DisablePackageSource(args.Name);
+            }
+
+            if (args.Verbosity != CommandLine.Verbosity.Quiet)
+            {
+                if (enabled)
+                {
+                    Console.WriteLine(string.Format(CultureInfo.CurrentCulture,
+                        Strings.SourcesCommandSourceEnabledSuccessfully, args.Name));
+                }
+                else
+                {
+                    Console.WriteLine(string.Format(CultureInfo.CurrentCulture,
```

Comment:
Same applies everywhere.

Reply:
yep. no console....even though console is your firend.